pkg/lib/gatherer/gatherer.go

```diff
@@ -90,3 +108,52 @@ func ListInstalledIstioRevisions(ctx context.Context, kubeclient client.Client)
 
 	return istioRevisionVersions, nil
 }
+
+func GetIstioPodsVersion(ctx context.Context, kubeClient client.Client) (string, error) {
+	pods, err := ListIstioCPPods(ctx, kubeClient)
+	if err != nil {
+		return "", err
+	}
+	currentVersion := &NoVersion
+	containersToCheck := []string{"discovery", "istio-proxy", "install-cni"}
+	for _, pod := range pods.Items {
+		if pod.ObjectMeta.DeletionTimestamp != nil {
+			continue
+		}
+		for _, container := range pod.Spec.Containers {
+			if !slices.Contains(containersToCheck, container.Name) {
+				continue
+			}
+			version, err := getImageVersion(container.Image)
+			if err != nil {
+				return "", err
+			}
+			if currentVersion.Compare(&NoVersion) == 0 {
+				currentVersion = version
+				continue
+			} else if currentVersion.Compare(version) != 0 {
+				return "", fmt.Errorf("Image version of pod %s: %s do not match version: %s", pod.Name, version.String(), currentVersion.String())
+			}
+		}
+	}
+	if currentVersion.Compare(&NoVersion) == 0 {
+		return "", errors.New("Unable to obtain installed Istio image version")
+	}
+	return currentVersion.String(), nil
+}
+
+func getImageVersion(image string) (*semver.Version, error) {
+	matches := reference.ReferenceRegexp.FindStringSubmatch(image)
+	if matches == nil || len(matches) < 3 {
+		return &NoVersion, fmt.Errorf("Unable to parse container image reference: %s", image)
+	}
+	imageTag := matches[2]
+	if strings.Contains(imageTag, "-") {
```

Comment:
Maybe we can add a short comment why this handling of tags that contain "-" is necessary as it wasn't so obvious for me at the first sight.

Reply:
Ohhh nooo, you are asking me to add comments in code 😬

Reply:
will try to refactor to make it more clear :)